### ISSUE TITLE
Any way to pass a callback to beforeSaveCell function?

### ISSUE THREAD

**Issue by CoralSilver** [AllenFang/react-bootstrap-table#1245]
I want to implement a confirmation modal and block the beforeSaveCell function until the user has hit OK or Cancel button. Is there a way to do this?

**Comment by AllenFang** [comment]
Use `options.handleConfirmDeleteRow`
```js
const options = {
  handleConfirmDeleteRow: function(cb, dropRowKeys) {
    //.... ask user here
    if (ok) cb();
    else .....
  }
}
```

**Comment by CoralSilver** [comment]
Do you mean I should use handleConfirmDeleteRow instead of beforeSaveCell for cell editing? Do I need to include a function in this.cellEditProp?

**Comment by AllenFang** [comment]
ah.... shit, my bad. please ignore it, sorry.

For `beforeSaveCell`, we dont have this kind of feature but I think maybe I can support it like `options.handleConfirmDeleteRow`

**Comment by CoralSilver** [comment]
Just to be clear I'm trying to implement a custom confirm modal not the browser confirm.

**Comment by CoralSilver** [comment]
That would be great. So you can't pass a can't pass a callback into validator then either right?

**Comment by AllenFang** [comment]
um~
my solution is just like `options.handleConfirmDeleteRow`

```js
const options = {
  beforeSaveCell: function(rowIndex, fieldName, value, cb) {
    //open your model and pass cb to your application
   return "SOME_SPECIFIED_VALUE";
  }
}

// if user confirm this editing, call the `cb`
```

is that ok for you?

**Comment by CoralSilver** [comment]
The function doesn't seem to be firing. Do I have to specify prop for cellEditProp beforeSaveCell still?

**Comment by AllenFang** [comment]
> The function doesn't seem to be firing

of course, because I doesn't implement it, I just want to confirm with you

**Comment by CoralSilver** [comment]
Oh sorry. I didn't understand. I'm trying to call the confirm modal using this plugin the[ React Confirm plugin](https://github.com/haradakunihiko/react-confirm) which returns a promise. With your proposed solution can I do something like this?: 

```
const options = {
beforeSaveCell: function(rowIndex, fieldName, value, this.confirmEdit) {
        let toSave = false;
        confirm('Are you sure you want to edit this value?').then(
        (result) => {
          // `proceed` callback
          console.log('proceed called');
          toSave = true;
        },
        (result) => {
          // `cancel` callback
          console.log('cancel called');
          toSave = false;
        }
        );
       // save change to database if toSave is true and update cell, otherwise return false and have cell reset
        this.confirmEdit(rowIndex, fieldName, value, toSave);
}
```

**Comment by AllenFang** [comment]
@CoralSilver, got it, I'll try it, if it work, I'll release in near future. Anyway, I guess it can work well :)

**Comment by AllenFang** [comment]
@CoralSilver, sorry about this issue, after I trying, I found there's a little bit complicated to implement this feature. so may not be implement in near future.

**Comment by AllenFang** [comment]
Fixed on `v3.3.1`, and check https://github.com/AllenFang/react-bootstrap-table/blob/master/examples/js/cell-edit/cell-edit-hook-table.js#L42 to learn how to use it.

let me know if you still got any questions, thanks.

**Comment by CoralSilver** [comment]
@AllenFang  Thanks so much for making that change. It works perfectly with React-Confirm.
